xpdacq/xpdacq.py

```diff
@@ -386,9 +386,15 @@ def __call__(self, sample, plan, subs=None, *,
             raise ValueError("These keys in metadata_kw are illegal "
                              "because they are always in sample: "
                              "{}".format(set(sample) & set(metadata_kw)))
-        if self._beamtime.get('bt_wavelength') is None:
-            print("WARNING: there is no wavelength information in current"
-                  "beamtime object, scan will keep going....")
+        if self._beamtime.wavelength is None:
+            warnings.warn("no wavelength is set to current beamtime "
+                          "object. Obtain the correct wavelength and "
+                          "run:\n>>> {}\n"
+                          "to make a auto-reduction work properly."
+                          .format('bt.wavelength = <value>'),
+                          UserWarning)
```

Comment:
To turn off this warning for the current beamtime, type <???> at the collection prompt.